datadog.go

```diff
@@ -18,75 +17,60 @@ const (
 )
 
 // NewExporter exports to a datadog client
-func NewExporter(opts Options) (*Exporter, error) {
-	if opts.StatsAddr == "" {
-		opts.StatsAddr = DefaultStatsAddrUDP
+func NewExporter(cfg Config, options ...statsd.Option) (*Exporter, error) {
+	if cfg.StatsAddr == "" {
+		cfg.StatsAddr = DefaultStatsAddrUDP
 	}
-	if opts.MetricNameFormatter == nil {
-		opts.MetricNameFormatter = defaultFormatter
-	}
-	client, err := statsd.New(opts.StatsAddr)
+	client, err := statsd.New(cfg.StatsAddr, options...)
 	if err != nil {
 		return nil, err
 	}
 	return &Exporter{
 		client: client,
-		opts:   opts,
+		cfg:    cfg,
 	}, nil
 }
 
 // Options contains options for configuring the exporter.
-type Options struct {
+type Config struct {
 	// StatsAddr specifies the host[:port] address for DogStatsD. It defaults
 	// to localhost:8125.
 	StatsAddr string
 
-	// Tags specifies a set of global tags to attach to each metric.
-	Tags []string
-
 	// UseDistribution uses a DataDog Distribution type instead of Histogram
 	UseDistribution bool
-
-	// MetricNameFormatter lets you customize the metric name that gets sent to
-	// datadog before exporting
-	MetricNameFormatter func(namespace, name string) string
```

Comment:
Let's put this back, because this is more business logic oriented: for example, sometimes people want to "capitalize" all metric names, and those metric names are coming from a library that they cannot override. 

I'm still okay with removing the string formatter below (aka sanitizeString) as long as we know that the official OpenCensus exporter didn't need to do this either, maybe we can open an issue on their repo asking them why they did it and not let the agent do the "correct work"?

Reply:
I think I'll leave it out for now and maintain a fork. I don't think business logic like capitalizing metric names is something I want to support, and certainly not within the exporter as this is too late in the pipeline. I may revisit this later when I understand the pipeline better for other things like dropping instrumented tags I wish to exclude from export.

Reply:
Sounds good